test/connectors/ConnectorSpec.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock._
+import com.github.tomakehurst.wiremock.http.Fault
+import com.github.tomakehurst.wiremock.matching.RequestPatternBuilder
+import com.github.tomakehurst.wiremock.stubbing.StubMapping
+import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.Application
+import play.api.http.{HeaderNames, MimeTypes, Status}
+import play.api.inject.guice.{GuiceApplicationBuilder, GuiceableModule}
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.ExecutionContext
+
+trait ConnectorSpec
+    extends AnyWordSpec
+    with GuiceOneAppPerSuite
+    with Status
+    with HeaderNames
+    with MimeTypes
+    with Matchers
+    with ScalaFutures
+    with IntegrationPatience {
+
+  implicit val hc: HeaderCarrier         = HeaderCarrier()
+  implicit lazy val ec: ExecutionContext =
+    scala.concurrent.ExecutionContext.global //TODO: remove lazy keyword when Caching spec is done.
```

Comment:
Is the TODO to be resolved as part of this ticket?

Reply:
Nope this isnt part of this ticket. But needs to be fixed though.